src/lib_main.rs

```diff
@@ -0,0 +1,327 @@
+///
+/// kanata remaps key presses to other keys or complex actions depending on the
+/// configuration for that key. You can find the guide for creating a config
+/// file here: https://github.com/jtroo/kanata/blob/main/docs/config.adoc
+///
+/// If you need help, please feel welcome to create an issue or discussion in
+/// the kanata repository: https://github.com/jtroo/kanata
+struct Args {
+    // Display different platform specific paths based on the target OS
+    #[cfg_attr(
+        target_os = "windows",
+        doc = r"Configuration file(s) to use with kanata. If not specified, defaults to
+kanata.kbd in the current working directory and
+'C:\Users\user\AppData\Roaming\kanata\kanata.kbd'"
+    )]
+    #[cfg_attr(
+        target_os = "macos",
+        doc = "Configuration file(s) to use with kanata. If not specified, defaults to
+kanata.kbd in the current working directory and
+'$HOME/Library/Application Support/kanata/kanata.kbd.'"
+    )]
+    #[cfg_attr(
+        not(any(target_os = "macos", target_os = "windows")),
+        doc = "Configuration file(s) to use with kanata. If not specified, defaults to
+kanata.kbd in the current working directory and
+'$XDG_CONFIG_HOME/kanata/kanata.kbd'"
+    )]
+    #[arg(short, long, verbatim_doc_comment)]
+    cfg: Option<Vec<PathBuf>>,
+
+    /// Port or full address (IP:PORT) to run the optional TCP server on. If blank, no TCP port will be
+    /// listened on.
+    #[cfg(feature = "tcp_server")]
+    #[arg(
+        short = 'p',
+        long = "port",
+        value_name = "PORT or IP:PORT",
+        verbatim_doc_comment
+    )]
+    tcp_server_address: Option<SocketAddrWrapper>,
+    /// Path for the symlink pointing to the newly-created device. If blank, no
+    /// symlink will be created.
+    #[cfg(target_os = "linux")]
+    #[arg(short, long, verbatim_doc_comment)]
+    symlink_path: Option<String>,
+
+    /// List the keyboards available for grabbing and exit.
+    #[cfg(target_os = "macos")]
+    #[arg(short, long)]
+    list: bool,
+
+    /// Enable debug logging.
+    #[arg(short, long)]
+    debug: bool,
+
+    /// Enable trace logging; implies --debug as well.
+    #[arg(short, long)]
+    trace: bool,
+
+    /// Remove the startup delay on kanata.
+    /// In some cases, removing the delay may cause keyboard issues on startup.
+    #[arg(short, long, verbatim_doc_comment)]
+    nodelay: bool,
+
+    /// Milliseconds to wait before attempting to register a newly connected
+    /// device. The default is 200.
+    ///
+    /// You may wish to increase this if you have a device that is failing
+    /// to register - the device may be taking too long to become ready.
+    #[cfg(target_os = "linux")]
+    #[arg(short, long, verbatim_doc_comment)]
+    wait_device_ms: Option<u64>,
+
+    /// Validate configuration file and exit
+    #[arg(long, verbatim_doc_comment)]
+    check: bool,
+}
+
+/// Parse CLI arguments and initialize logging.
+fn cli_init() -> Result<ValidatedArgs> {
```

Comment:
I appreciate the effort to reuse code here, but I would rather the changes to existing code be minimized in an already-large PR.

Related to the comment about undoing the refactor to a new file, I would rather there be duplicated code with a separate version/variant (doesn't need to keep to the same name) of `cli_init` and `main_impl` in a GUI-specific file.

Reply:
moved